sw/device/lib/testing/check.h

```diff
@@ -72,4 +73,34 @@
     }                                                                         \
   } while (false)
 
+/**
+ * Checks that the given DIF call returns kDifOk. If the DIF call returns a
+ * different dif_result_t value (defined in sw/device/lib/dif/dif_base.h), this
+ * function logs and then aborts.
+ *
+ * @param DIF call to invoke and check its return value .
+ * @param ... arguments to a LOG_* macro, which are evaluated if the check
+ * fails.
+ */
+#define CHECK_DIF_OK(dif_call, ...)                       \
+  do {                                                    \
+    if (dif_call != kDifOk) {                             \
+      /* NOTE: because the condition in this if           \
+         statement can be statically determined,          \
+         only one of the below string constants           \
+         will be included in the final binary.*/          \
+      if (GET_NUM_VARIABLE_ARGS(_, ##__VA_ARGS__) == 0) { \
+        LOG_ERROR("CHECK-fail: " #dif_call);              \
```

Comment:
Minor nit: perhaps `("DIF invocation failed: " #dif_call)`?

Reply:
Fixed. Made even more concise with "DIF-fail: ..." since printing over UART is expensive performance-wise.